sklearn/ensemble/forest.py

```diff
@@ -69,6 +69,22 @@ class calls the ``fit`` method of each sub-estimator on random samples
 
 MAX_INT = np.iinfo(np.int32).max
 
+def _generate_sample_indices(tree, n_samples):
+    """Private function used to _parallel_build_trees function."""
+    random_state = check_random_state(tree.random_state)
+    sample_indices = random_state.randint(0, n_samples, n_samples)
+
+    return sample_indices
+
+def _generate_unsampled_indices(tree, n_samples):
```

Comment:
sure.
I'll change to pass random_state, not tree.
(I didn't think carefully, there is just copied code)